src/common/window/session-storage.js

```diff
@@ -6,34 +6,68 @@
  * @environment Browser script
  */
 import { generateRandomHexString } from '../ids/unique-id'
-import { isBrowserScope } from '../util/global-scope'
 
-export { getCurrentSessionIdOrMakeNew }
-
-const SESS_ID = 'NRBA_SESSION_ID' // prevents potential key collisions in session storage
+const AGENT_PREFIX = 'NRBA/'
 
 /**
- * @returns {string} This tab|window's session identifier, or a new ID if not found in storage
+ * @param {string} key
+ * @param {string} value
+ * @returns true, if item was set successfully. false, if storage is full or setItem threw an exception.
  */
-function getCurrentSessionIdOrMakeNew () {
-  if (!isBrowserScope) {
-    return null
+export function putInBrowserStorage (key, value, subgroup = '') {
+  try {
+    window.sessionStorage.setItem(AGENT_PREFIX + subgroup + key, value)
+    return true
+  } catch (ex) {
+    // Session storage may be full.
+    return false
   }
+}
+/**
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem}
+ * @param {string} key
+ */
+function getFromBrowserStorage (key, subgroup = '') {
+  return window.sessionStorage.getItem(AGENT_PREFIX + subgroup + key)
+}
+export function removeFromBrowserStorage (key, subgroup = '') {
+  return window.sessionStorage.removeItem(AGENT_PREFIX + subgroup + key)
+}
```

Comment:
These function will need `try..catch` blocks. Safari throws an exception any time you try to use `sessionStorage` in a private browser window.

<img width="333" alt="image" src="https://user-images.githubusercontent.com/3984979/230088493-d7b693ab-4cae-4259-bc6c-a158d622c9ed.png">

Reply:
Esp, `removeFromBrowserStorage` which doesn't have a catch in outer parts, thanks.